### PR TITLE
Treat invokeinterface bridges similarly to invokespecial

### PR DESCRIPTION
**Pull request by cushon**
In both cases we need to avoid using invokesuper. For interface bridges,
using invokesuper will fail since the method being bridged to is in a
superinterface, not a superclass.

Starting in Java 8, javac emits default bridge methods in interfaces,
which use invokeinterface to bridge to the target method.

Without this fix, the included test case fails with:

```
Caused by: java.lang.NoSuchMethodError: java.lang.Object.f(Ljava/lang/Object;)V
    at B$$EnhancerByCGLIB$$7aa3e8e0.f(<generated>)
    ... 28 more
```

### CODE REVIEW

File: cglib/src/main/java/net/sf/cglib/proxy/BridgeMethodResolver.java [hunk 1]
```diff
@@ -32,9 +32,9 @@
 import org.objectweb.asm.Opcodes;
 
 /**
- * Uses bytecode reflection to figure out the targets of all bridge methods
- * that use invokespecial, so that we can later rewrite them to use invokevirtual.
- * 
+ * Uses bytecode reflection to figure out the targets of all bridge methods that use invokespecial
+ * and invokeinterface, so that we can later rewrite them to use invokevirtual.
```
Comment (sameb) [inline cglib/src/main/java/net/sf/cglib/proxy/BridgeMethodResolver.java]: can you clarify this to explain why we need to find invokeinterface (basically the description from the PR)?

File: cglib/src/test/java/net/sf/cglib/proxy/TestEnhancer.java [hunk 2]
```diff
@@ -1418,6 +1418,141 @@ public Class findClass(String name) throws ClassNotFoundException {
...
+        // }
+        // interface B<T extends Number> extends A<T>{
+        //     void f(T t);
+        // }
+
+        final Map<String, byte[]> classes = new HashMap<String, byte[]>();
+        {
+            ClassWriter classWriter = new ClassWriter(0);
+            classWriter.visit(
+                    Opcodes.V1_8,
```
Comment (sameb) [inline cglib/src/test/java/net/sf/cglib/proxy/TestEnhancer.java]: we test w/ java1.7 too.    could you maybe add a variant of this that's based off the classes as compiled from source, and optionally keep this but bail if < java8?

**Comment by cushon** [comment]
> can you clarify this to explain why we need to find invokeinterface (basically the description from the PR)?

done

> could you maybe add a variant of this that's based off the classes as compiled from source

The tests are compiled with `-source 6 -target 6`, and the test case relies on Java 8 features. I'm not sure how to add a variant that uses a newer language level?

> and optionally keep this but bail if < java8?

done

> any idea how/why this only became a problem after 942ab9e?

I'm not sure it was cauesd by 942ab9e. I ran into this trying to use a newer version of cglib with mockito 1.x, which is currently using cglib 2.2.: https://github.com/mockito/mockito/commit/b6b13e8b28d163f989e4f95c941a3da5c619d770

**Comment by cushon** [comment]
I think force pushing confused it. I started a new PR: #142

**Pull request closed without merging**